src/sentry/models/projectsdk.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from enum import Enum
+
+from django.db import models
+from packaging.version import InvalidVersion
+from packaging.version import parse as parse_version
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import BoundedIntegerField, FlexibleForeignKey, region_silo_model, sane_repr
+from sentry.db.models.base import DefaultFieldsModel
+from sentry.models.project import Project
+from sentry.utils import metrics
+from sentry.utils.cache import cache
+from sentry.utils.hashlib import md5_text
+
+
+class EventType(Enum):
+    PROFILE = 0
+    PROFILE_CHUNK = 1
+
+    @classmethod
+    def as_choices(cls) -> Sequence[tuple[int, str]]:
+        return (
+            (cls.PROFILE.value, "profile"),
+            (cls.PROFILE_CHUNK.value, "profile_chunk"),
+        )
+
+
+@region_silo_model
+class ProjectSDK(DefaultFieldsModel):
+    __relocation_scope__ = RelocationScope.Organization
+
+    date_updated = models.DateTimeField(auto_now=True)
+
+    project = FlexibleForeignKey("sentry.Project", on_delete=models.CASCADE)
+    event_type = BoundedIntegerField(choices=EventType.as_choices())
+    sdk_name = models.CharField()
+    sdk_version = models.CharField()
+
+    class Meta:
+        unique_together = (("project", "event_type", "sdk_name"),)
+
+    __repr__ = sane_repr("project", "event_type", "sdk_name", "sdk_version")
+
+    @classmethod
+    def get_cache_key(cls, project: Project, event_type: EventType, sdk_name: str):
+        return f"projectsdk:1:{project.id}:{event_type.value}:{md5_text(sdk_name).hexdigest()}"
+
+    @classmethod
+    def update_with_newest_version_or_create(
+        cls,
+        project: Project,
+        event_type: EventType,
+        sdk_name: str,
+        sdk_version: str,
+    ) -> ProjectSDK:
+        with metrics.timer(
+            "models.projectsdk.update_with_newest_version_or_create"
+        ) as metrics_tags:
```

Comment:
You might consider putting a lock around this too, in addition to the cache. There's a thundering herd problem here where once the cache expires, for high volume projects we'll hit postgres multiple times quickly. If you use a lock, you can just catch the unable to acquire exception and continue on, since the other event can happily update the row.

You can lock on the same key as the hash, maybe just with a different namespace